src/coreclr/vm/riscv64/asmconstants.h

```diff
@@ -250,5 +250,34 @@ ASMCONSTANTS_C_ASSERT(CallCountingStubData__TargetForMethod == offsetof(CallCoun
 #define CallCountingStubData__TargetForThresholdReached 0x10
 ASMCONSTANTS_C_ASSERT(CallCountingStubData__TargetForThresholdReached == offsetof(CallCountingStubData, TargetForThresholdReached))
 
+#ifdef PROFILING_SUPPORTED
+#define PROFILE_ENTER    1
+#define PROFILE_LEAVE    2
+#define PROFILE_TAILCALL 4
+
+#define SIZEOF__PROFILE_PLATFORM_SPECIFIC_DATA 312
+#define PROFILE_PLATFORM_SPECIFIC_DATA__argumentRegisters 16
+#define PROFILE_PLATFORM_SPECIFIC_DATA__functionId 80
+#define PROFILE_PLATFORM_SPECIFIC_DATA__floatArgumentRegisters 88
+#define PROFILE_PLATFORM_SPECIFIC_DATA__probeSp 152
+#define PROFILE_PLATFORM_SPECIFIC_DATA__profiledSp 160
+#define PROFILE_PLATFORM_SPECIFIC_DATA__hiddenArg 168
+#define PROFILE_PLATFORM_SPECIFIC_DATA__flags 176
+
+ASMCONSTANTS_C_ASSERT(SIZEOF__PROFILE_PLATFORM_SPECIFIC_DATA == sizeof(PROFILE_PLATFORM_SPECIFIC_DATA))
+
+#define ASMCONSTANTS_C_ASSERT_OFFSET(type, field) \
+    ASMCONSTANTS_C_ASSERT(type##__##field == offsetof(type, field))
+ASMCONSTANTS_C_ASSERT_OFFSET(PROFILE_PLATFORM_SPECIFIC_DATA, argumentRegisters)
+ASMCONSTANTS_C_ASSERT_OFFSET(PROFILE_PLATFORM_SPECIFIC_DATA, functionId)
+ASMCONSTANTS_C_ASSERT_OFFSET(PROFILE_PLATFORM_SPECIFIC_DATA, floatArgumentRegisters)
+ASMCONSTANTS_C_ASSERT_OFFSET(PROFILE_PLATFORM_SPECIFIC_DATA, probeSp)
+ASMCONSTANTS_C_ASSERT_OFFSET(PROFILE_PLATFORM_SPECIFIC_DATA, profiledSp)
+ASMCONSTANTS_C_ASSERT_OFFSET(PROFILE_PLATFORM_SPECIFIC_DATA, hiddenArg)
+ASMCONSTANTS_C_ASSERT_OFFSET(PROFILE_PLATFORM_SPECIFIC_DATA, flags)
+#undef ASMCONSTANTS_C_ASSERT_OFFSET
+
+#endif // PROFILING_SUPPORTED
+
```

Comment:
It does not sync with other plaforms. IMO, I think it is better to update with all other platforms in this case. @jkotas

Reply:
Like in the other comment on `cgencpu.h`, IMO having the field offsets pinned to the C struct with static asserts is a net improvement over hardcoded offsets in asm.

I wanted to limit the impact of the change to RISC-V since this PR is large enough. But I can do the same for other platforms if need be.

Reply:
Okay. I understand your concerns. However, I prefer that the same structures and functions should be in the same file for all platforms as much as possible. So I want you to refactoring all the other platforms in another PR. (or you can focus bug-fix only in this PR. Then updates all (including RISC-V) in another PR.) Thank you.
@jkotas Could you give any comment about this?

Reply:
I agree that it would be nice to use named constants that are validated at compile time, across all platforms. This cleanup can be done in a separate PR.

Reply:
OK, I'll do an analogous cleanup for `PROFILE_PLATFORM_SPECIFIC_DATA` in a separate PR.